Add an EventQueue-based blinky to explain flow control

LDong-Arm opened the pull request:
This is needed for https://github.com/ARMmbed/mbed-os-5-docs/pull/1320

hugueskamba commented on Tutorials_UsingAPIs/Flow-Control-EventQueue/main.cpp:
```diff
@@ -0,0 +1,27 @@
+ */
+
+#include "mbed.h"
+#include "mbed_events.h"
```
This is already included in "mbed.h".
If you insist on re-including here, it is a good idea to include with the directory to give some kind of namespace (i.e `#include "events/mbed_events.h"`).

LDong-Arm replied:
I'll change it. It was copied from another snippet.

hugueskamba commented on Tutorials_UsingAPIs/Flow-Control-EventQueue/main.cpp:
```diff
@@ -0,0 +1,27 @@
+#include "mbed.h"
+#include "mbed_events.h"
+
+DigitalOut led1(LED1);
```
What is the reason for having these two objects as global? Why are they not created within `main()`?

LDong-Arm replied:
I think in Mbed OS applications it's common to put "hardware" objects like buttons, LEDs, network interface etc. in static memory, as in most examples.

ithinuel replied:
It is still generally considered bad practice though. We should not encourage that.

LDong-Arm replied:
For this example I can move them into `main()`. Though it'll make it inconsistent with other examples/snippets we have.

LDong-Arm replied:
Done

hugueskamba commented on Tutorials_UsingAPIs/Flow-Control-EventQueue/main.cpp:
```diff
@@ -0,0 +1,27 @@
+    EventQueue queue;
+
+    // events are simple callbacks
+    queue.call_every(1000, [] {
```
The follow seems clearer to me (for readability):
```c++
queue.call_every(1000, [] {led1 = !led1;});
```
It is still less than 80 chars wide. But it's your call.

LDong-Arm replied:
I'll change it

LDong-Arm replied:
I just changed this to a non-lambda callback, because lambda isn't used anywhere yet in our documentation

LDong-Arm commented:
@hugueskamba Thanks for the review, I just addressed your comment and changed EventQueue callbacks to non-lambda in order to be consistent with existing docs

jamesbeyond commented:
@LDong-Arm could you rebase, the CI build should be successful now.

LDong-Arm commented:
@jamesbeyond I've rebased the branch.
@hugueskamba @ithinuel Thanks for the review, I've addressed your comments.

jamesbeyond commented:
> @jamesbeyond I've rebased the branch.
> @hugueskamba @ithinuel Thanks for the review, I've addressed your comments.

astyle issue, please fix it in your code

LDong-Arm commented:
> astyle issue, please fix it in your code

Thanks, done.

jamesbeyond commented:
Is this one good to be merged? @LDong-Arm

LDong-Arm commented:
> Is this one good to be merged? @LDong-Arm

Yes, if @hugueskamba @ithinuel don't have additional remarks?
